var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -37,7 +37,8 @@ class Julia(Package):
             git='https://github.com/JuliaLang/julia.git', branch='master')
     version('release-0.5',
             git='https://github.com/JuliaLang/julia.git', branch='release-0.5')
-    version('0.5.2', '8c3fff150a6f96cf0536fb3b4eaa5cbb', preferred=True)
+    version('0.6.2', '255d80bc8d56d5f059fe18f0798e32f6', preferred=True)        
```

Comment:
For consistency put this line above the `'release-0.5'` line. Out of curiosity, how is spack actually handling this? Would `'release-0.5'` be preferred over `0.6.2`? If not you can remove the `preferred=True`

Reply:
> Out of curiosity, how is spack actually handling this?

When determining which version to install, Spack doesn't care about the order of versions in the `package.py` file. It is just a convention to sort things from newest to oldest. The default version is always the `preferred` version, unless overridden in `packages.yaml`. If there is no `preferred=True`, the default is the newest numbered release (so ignoring `master` and `develop`). You can always see the order of releases by running `spack info <packagename>`.

Reply:
thanks @adamjstewart just checked `spack info julia` the `release-xx` versions are below the lowest numbered version, i.e. you won't need the `preferred=True` at all